### PR TITLE
Add support for rename to fake redis.

### PR DESCRIPTION
**Pull request by hodgestar**
It's 2016 and we're still playing this game. ;)

### CODE REVIEW

**Comment by hodgestar** [comment]
@KaitCrawford ready for review in the morning.

**Comment by KaitCrawford** [comment]
@hodgestar looks good to my untrained eye. :+1: 
@justinvdm or @rudigiesler, if you're happy then I'll merge and we can work on releasing it

File: vumi/persist/fake_redis.py [hunk 1]
```diff
@@ -220,6 +220,16 @@ def delete(self, key):
             self._known_key_existence[key] = False
         return existed
 
+    @maybe_async
+    def rename(self, key, newkey):
+        if key == newkey:
+            raise ResponseError("source and destination objects are the same")
```
Comment (justinvdm) [inline vumi/persist/fake_redis.py]: Real redis seems okay with this:
```redis> set a 23
OK
redis> rename a a
OK
```

Comment (justinvdm) [inline vumi/persist/fake_redis.py]: Ah, so the test console in [redis.io](redis.io)'s docs is okay with this, but running this locally (v3.0.5), I get the error.

**Comment by justinvdm** [comment]
:+1: from me too, happy with this landing

**Pull request merged**